src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2471,36 +2468,20 @@ def groupby_ngroups(
         level = groupby_kwargs.get("level", None)
         dropna = groupby_kwargs.get("dropna", True)
 
-        can_be_distributed = check_is_groupby_supported_by_snowflake(by, level, axis)
-
-        def fallback_ngroups() -> int:
-            """
-            Creates a SnowflakeQueryCompiler through a fallback operation,
-            whose snowpark dataframe holds the result of the ngroups operation.
-            The snowpark dataframe has the form of [Row('0'=<ngroups_value>, ...)]
-            and we call collect to return this result. Please note that this will
-            trigger an eager evaluation.
-            """
-            query_compiler: SnowflakeQueryCompiler = GroupByDefault.register(
-                native_pd.core.groupby.DataFrameGroupBy.ngroups
-            )(
-                self,
-                by=by,
-                axis=axis,
-                groupby_kwargs=groupby_kwargs,
+        is_supported = check_is_groupby_supported_by_snowflake(by, level, axis)
+        if not is_supported:
+            ErrorMessage.not_implemented(
+                "GroupBy.ngroups is not implemented for the given parameter combination"
             )
-            ngroups_result = query_compiler._modin_frame.ordered_dataframe.collect()
-            return ngroups_result[0]["0"]
-
-        if not can_be_distributed:
-            return fallback_ngroups()
 
         query_compiler = get_frame_with_groupby_columns_as_index(
             self, by, level, dropna
         )
 
         if query_compiler is None:
-            return fallback_ngroups()
+            ErrorMessage.not_implemented(
+                "GroupBy.ngroups is not implemented for the given parameter combination"
```

Comment:
If we can also tell users which exact parameter is not supported. That would be more actionable information. Not a pupr blocker though so If that leads to much bigger refactoring we can as follow up after PuPr..

Reply:
For that, we should need to refactor `check_is_groupby_supported_by_snowflake` and `check_is_aggregation_supported_in_snowflake` which is used in a lot of places. I think it's better to do that in a follow-up PR if necessary given the CHANGELOG/our documentation documents unsupported params.

Reply:
We could get some info from the code path without changing the code right? Some errors raised below may specific to some usage cases. Can we make "for the given parameter combination" more concrete without changing code here?

Reply:
Oh, I see you already have the group method name included. Then it should be good for now.

Reply:
The alternative is something like "`Groupby.{agg_name}` is not implemented for `pd.Grouper`, `axis == 1`, both `by` and `level` are configured, `by` contains any non-pandas hashable labels, or unsupported aggregation parameters."

Reply:
I like detailed error message.